trinity/extensibility/plugin_manager.py

```diff
@@ -116,65 +190,72 @@ def broadcast(self, event: BaseEvent, exclude: BasePlugin = None) -> None:
 
-    def _is_responsible_for_plugin(self, plugin: BasePlugin) -> bool:
-
-        main_or_isolated_plugin = plugin.process_scope in self.MAIN_AND_ISOLATED_SCOPES
-        shared_plugin = not main_or_isolated_plugin
-
-        manager_for_main_or_isolated = isinstance(self._scope, MainAndIsolatedProcessScope)
-        manager_for_shared = not manager_for_main_or_isolated
-
-        return ((main_or_isolated_plugin and manager_for_main_or_isolated) or
-                (shared_plugin and manager_for_shared))
-
-    def _create_context_for_plugin(self,
-                                   plugin: BasePlugin,
-                                   args: Namespace,
-                                   chain_config: ChainConfig,
-                                   boot_kwargs: Dict[str, Any]) -> PluginContext:
-
-        context: PluginContext = None
-        if plugin.process_scope in self.MAIN_AND_SHARED_SCOPES:
-            # A plugin that runs in a shared process as well as a plugin that overtakes the main
-            # process uses the endpoint of the PluginManager which will either be the main
-            # endpoint or the networking endpoint in the case of Trinity
-            context = PluginContext(self._scope.endpoint)
-        elif plugin.process_scope is PluginProcessScope.ISOLATED:
-            # A plugin that runs in it's own process gets a new endpoint created to get
-            # passed into that new process
-
-            # mypy doesn't know it can only be that scope at this point. The `isinstance`
-            # check avoids adding an ignore
-            if isinstance(self._scope, MainAndIsolatedProcessScope):
-                endpoint = self._scope.event_bus.create_endpoint(plugin.name)
-                context = PluginContext(endpoint)
-        else:
-            Exception("Invariant: unreachable code path")
+    async def shutdown(self) -> None:
+        """
+        Asynchronously shut down all started plugins.
+        """
+
+        if isinstance(self._scope, MainAndIsolatedProcessScope):
+            raise UnsuitableShutdownError("Use `shutdown_blocking` for instances of this scope")
 
-        context.args = args
-        context.chain_config = chain_config
-        context.boot_kwargs = boot_kwargs
+        self._logger.info("Shutting down PluginManager with scope %s", type(self._scope))
 
-        return context
+        stop_results = await asyncio.gather(*self._stop_async_plugins(), return_exceptions=True)
+
+        for plugin, result in zip(self._started_plugins, stop_results):
+            if isinstance(result, Exception):
+                self._logger.error(
+                    'Exception thrown while stopping plugin %s: %s', plugin.name, result
+                )
+            else:
+                self._logger.info("Successfully stopped plugin: %s", plugin.name)
+
+    def _stop_async_plugins(self) -> Iterable[Awaitable[Optional[Exception]]]:
```

Comment:
I pushed this out into a private method rather than using an inline comprehension for two reasons:

1. I wanted to do `self._logger.info("Stopping plugin: %s", plugin.name)` to have the same logs as with sync stops
2. We need an `isinstance` check to avoid a cast